spec/features/merchants/merchant_sees_order_spec.rb

```diff
@@ -0,0 +1,121 @@
+require 'rails_helper'
+
+RSpec.describe "When I visit an order show page from my dashboard" do
+  before :each do
+    @merchant = create(:merchant)
+    @customer = create(:user, address: "123 Main St", city: "Denver", state: "CO", zipcode: 80302)
+    @other_merchant = create(:merchant)
+    @item_1 = create(:item, user: @merchant)
+    @item_2 = create(:item, user: @merchant)
+    @item_3 = create(:item, user: @merchant)
+    @item_4 = create(:item, user: @merchant)
+    @item_5 = create(:item, user: @other_merchant)
+    @item_6 = create(:item, user: @other_merchant)
+    @item_7 = create(:item, user: @other_merchant)
+    @order_1 = create(:order, user: @customer, status: 'pending')
+    @order_2 = create(:order, user: @customer, status: 'completed')
+    @order_item_1 = create(:order_item, order: @order_1, item: @item_1, quantity: 9)
+    @order_item_2 = create(:order_item, order: @order_1, item: @item_2, quantity: 2)
+    @order_item_3 = create(:order_item, order: @order_1, item: @item_3, quantity: 7)
+    @order_item_4 = create(:order_item, order: @order_1, item: @item_4, quantity: 4)
+    @order_item_5 = create(:order_item, order: @order_1, item: @item_5, quantity: 5)
+    @order_item_6 = create(:order_item, order: @order_1, item: @item_6, quantity: 6)
+    @order_item_7 = create(:order_item, order: @order_1, item: @item_7, quantity: 7)
+    @order_item_8 = create(:order_item, order: @order_2, item: @item_1, quantity: 5, fulfilled: true)
+
+  end
+
+  context "as a merchant" do
+
+    it "I see customer/s name, address" do
+      login_as(@merchant)
+      visit dashboard_path
+      click_link(@order_1.id)
+      expect(current_path).to eq(merchant_order_path(@order_1))
+
+      expect(page).to have_content("Customer: #{@customer.name}")
+      expect(page).to have_content("Address: #{@customer.address}")
+      expect(page).to have_content(@customer.city)
+      expect(page).to have_content(@customer.state)
+      expect(page).to have_content(@customer.zipcode)
+
+    end
+
+    it "I do not see items being purchased from other merchants" do
+      login_as(@merchant)
+      visit dashboard_path
+      click_link(@order_1.id)
+
+      expect(page).to_not have_content(@item_5.name)
+      expect(page).to_not have_content(@item_6.name)
+      expect(page).to_not have_content(@item_7.name)
+    end
+
+    it "I see items that are being purchased from me" do
+      login_as(@merchant)
+      visit dashboard_path
+      click_link @order_1.id
+
+      expect(page).to have_link(@item_1.name)
+      expect(page).to have_link(@item_2.name)
+      expect(page).to have_link(@item_3.name)
+      expect(page).to have_link(@item_4.name)
+
+      #
+      # click_link(@item_1.name)
```

Comment:
this is commented out because it was giving me a weird error that seems to be related to the items show page. Would love help here figuring out why though.

<img width="1212" alt="screen shot 2019-02-17 at 1 54 04 am" src="https://user-images.githubusercontent.com/13354855/52910595-edd13d80-3256-11e9-85ac-ffcbf7456567.png">

Reply:
I bet because the order_items aren't fulfilled or something, so avg fulfillment time is being called on nil. I bet you could put an extra conditional on it that only runs it if at least one order item is fulfilled else it says not yet fulfilled or zero or something.

Reply:
I think it has something to do with when you try to go to one of the item show pages. Maybe it's because you're trying to do a calculation for fulfillment_time on an item that has never been fulfilled? You might need to put some logic in the item show page view to render some sort of default value if an item has not been fulfilled yet.

Reply:
I didn't see Tim's comment until after I had left mine but I think we had the same idea!